internal/runtime/client/client.go

```diff
@@ -290,6 +296,16 @@ func (c *client) CallExtension(ctx context.Context, hook runtimecatalog.Hook, na
 		return errors.Errorf("ExtensionHandler %q does not match group %s, hook %s", name, hookGVH.Group, hookGVH.Hook)
 	}
 
+	// Compute whether the object the call is being made for matches the namespaceSelector
+	namespaceLabels, err := c.matchNamespace(ctx, registration.NamespaceSelector, forObject.GetNamespace())
+	if err != nil {
+		return errors.Errorf("ExtensionHandler %q namespaceSelector could not be resolved", name)
+	}
+	// If the object namespace isn't matched by the registration NamespaceSelector return without calling.
+	if !namespaceLabels {
```

Comment:
I think for CallExtension that's an error, while for CallAllExtension it's fine if we just skip.

Example: Topology mutation: ClusterClass configures external patching, ExtensionConfig.spec.namespaceSelector doesn't match => This should be an error